src/widget.ts

```diff
@@ -0,0 +1,91 @@
+import { Widget } from '@lumino/widgets';
+import { PromiseDelegate } from '@lumino/coreutils';
+import ImageEditor from 'tui-image-editor';
+import { DocumentRegistry } from '@jupyterlab/docregistry';
+
+function resizeEditor() {
+  var editor = document.querySelector('.tui-image-editor');
+  var container = document.querySelector('.tui-image-editor-canvas-container');
+  var height = (container as HTMLElement).style.maxHeight;
+
+  (editor as HTMLElement).style.height = height;
+}
+
+export class ImageEditorWidget extends Widget {
+    private _editor: any;
+    private context: DocumentRegistry.Context;
+    private _ready = new PromiseDelegate<void>();
+
+    get ready(): PromiseDelegate<void> {
+        return this._ready;
+    }
+
+    constructor(context: DocumentRegistry.Context, maxWidth: number, maxHeight: number, cornerSize: number, rotatingPointOffset: number) {
+        super({
+          node: document.createElement("div")
+        });
+        this.context = context;
+        this.node.appendChild(document.createElement("div")).className = "tui-image-editor";
+        this._loadImageEditor(maxWidth, maxHeight, cornerSize, rotatingPointOffset);
+        context.ready.then(async value => {
+            await this.ready.promise;
+            const base64Response = await fetch(`data:image/${context.contentsModel?.type};base64,${context.model.toString()}`);
+            const blob = await base64Response.blob();
+    
+            this.setContent(URL.createObjectURL(blob), context.path);
+        });
+        this._ready.resolve(void 0);
```

Comment:
The _ready_ promise should be resolved when the image is loaded not before loading it in the editor (because at this is what the user expect to have his image ready for edition). And if for any reason the image cannot be loaded, you should invalidate the _ready_ promise to report to the user: _Something went wrong when loading your image_. You can achieve this with:

```suggestion
        context.ready.then(async value => {
            const base64Response = await fetch(`data:image/${context.contentsModel?.type};base64,${context.model.toString()}`);
            const blob = await base64Response.blob();
    
            this.setContent(URL.createObjectURL(blob), context.path);
            this._ready.resolve(void 0);
        }).catch(reason => {
          this._ready.reject(`Fail to load the image for ${context.path}:\n${reason}.`)
        });
```